adr/0002-generate-extension-metadata-in-deployment-module.adoc

```diff
@@ -0,0 +1,123 @@
+* generated information which is only available to the extension plugin when it's running in the runtime module
+
+If so, the consumers will need to be updated to read the deployment module, or those informations should move to `quarkus-extension.properties` (if the consumer has no access to the deployment module).
+
+A secondary benefit of this proposal is reducing the size (slightly) of the extension's runtime artifacts.
+
+=== Transition plan
+
+We will need to allow a long transition period (probably one major release lifespan), in which
+
+* The registry looks in both the runtime and deployment jars for `META-INF/quarkus-extension.yaml`, and possibly also the new metadata artifact
+* The extension plugin tolerates being invoked from both `runtime` and `deployment` modules (so it's probably simplest to have one 'umbrella' plugin which handles both modules). When invoked, it will look in both the runtime and deployment modules for `src/META-INF/quarkus-extension.yaml`.
+* We still package up a (less complete) `quarkus-extension.yaml` into the runtime module, so that back-level tools (such as the CLI, and maybe IDEs) can find it
+
+image::images/0002-transition-flow.svg[a transitional process, with fallbacks]
+Source::https://excalidraw.com/#json=A7UUcn3Vme98So4xp3jZ6,bemZxMtdn5vLP9MgpZtwJw
+
+In other words, the deployment module becomes the new 'best source', but
+(a) not every extension will have the information in the new place, so tools need to look in the old place
+(b) not every tool will know about the new place, so extensions need to also package the information in the old place
+
+== Consequences
+
+What currently uses the `quarkus-extension.yaml` beyond the registry? Does this affect IDEs? Do users ever look into jars for the file?
+
+== Scenarios
+
+=== Extension owners
+The ADR will affect extensions differently, depending how coupled they are to the core repo.
+
+==== Quarkus core extensions
+
+We will need to patch each extension's `deployment` `pom.xml` to invoke the extension plugin, and move the `src/META-INF/quarkus-extension.yaml`. We can do this any time after (or with) the plugin changes.
+
+==== Quarkiverse extensions
+
+We can do an auto-update using @gastaldi's script to update the deployment pom to invoke the extension plugin, but this can only be applied to extensions which are building against a Quarkus release which has these changes in.
+
+==== Unmanaged extensions
+
+We will need to broadcast information about the changes. We should assume that the transition will be slow for these extensions.
+
+=== Tooling
+
+==== Platform descriptor generation
+
+The tooling which generates the platform descriptor JSON will need to be updated to look for the new source of this metadata.
+
+==== Registry
+
+For platform releases, the registry consumes the generated platform descriptors, so assuming that tooling is updated, there won't be much of a change for platform metadata processing on the registry side.
```

Comment:
afaics extensions in platform should have exact same impact as non-platform extensions.

is only when using platform to get info on these details things gets different.

we have to keep reminding ourselves that just because our main platform has a set of extensions in it there should not be anything technical stopping us or someone from outside making a platform that has a different composition.

Reply:
I think what this is trying to say is that the upload process to the registry is different for platform and non-platform extensions. But it's suggested text from @aloubyansky, so I'll let him comment, since he knows much more than  me about the area.